GVFS/GVFS/CommandLine/DiagnoseVerb.cs

```diff
@@ -115,27 +118,30 @@ protected override void Execute(GVFSEnlistment enlistment)
                         this.CopyLocalCacheData(archiveFolderPath, localCacheRoot, gitObjectsRoot);
 
                         // corrupt objects
-                        this.CopyAllFiles(enlistment.DotGVFSRoot, Path.Combine(archiveFolderPath, GVFSConstants.DotGVFS.Root), GVFSConstants.DotGVFS.CorruptObjectsName, copySubFolders: false);
-
-                        // service
-                        this.CopyAllFiles(
-                            Paths.GetServiceDataRoot(string.Empty),
-                            archiveFolderPath,
-                            this.ServiceName,
-                            copySubFolders: true);
-
-                        // upgrader
-                        this.CopyAllFiles(
-                            ProductUpgrader.GetUpgradesDirectoryPath(),
-                            archiveFolderPath,
-                            ProductUpgrader.LogDirectory,
-                            copySubFolders: true,
-                            targetFolderName: ProductUpgrader.UpgradeDirectoryName);
-                        this.LogDirectoryEnumeration(
-                            ProductUpgrader.GetUpgradesDirectoryPath(), 
-                            Path.Combine(archiveFolderPath, ProductUpgrader.UpgradeDirectoryName),
-                            ProductUpgrader.DownloadDirectory, 
-                            "downloaded-assets.txt");
+                        this.CopyAllFiles(enlistment.DotGVFSRoot, Path.Combine(archiveFolderPath, GVFSConstants.DotGVFS.Root), GVFSConstants.DotGVFS.CorruptObjectsName, copySubFolders: false);
+
+                        if (!GVFSPlatform.Instance.IsUnderConstruction)
+                        {
+                            // service
+                            this.CopyAllFiles(
+                                Paths.GetServiceDataRoot(string.Empty),
+                                archiveFolderPath,
+                                this.ServiceName,
+                                copySubFolders: true);
+
+                            // upgrader
+                            this.CopyAllFiles(
```

Comment:
Maybe add a `GVFSPlatform.SupportsGVFSUpgrade`? And actually the `UpgradeVerb` code would benefit from that too, since it's still incorrectly checking the OS instead of a platform capability. cc @alameenshah in case he already has a fix in progress for that.

Reply:
I'll make this change